src/util/lie/se3.hpp

```diff
@@ -34,19 +57,19 @@ class SE3 {
 
     static void pushToTfTree(tf2_ros::TransformBroadcaster& broadcaster, std::string const& childFrameId, std::string const& parentFrameId, SE3 const& tf);
 
-    SE3() = default;
-
-    SE3(Eigen::Vector3d position, Eigen::Quaterniond const& rotation);
+    SE3(R3 const& position, SO3 const& rotation = {});
 
-    [[nodiscard]] SE3 applyLeft(SE3 const& transform);
+    template<typename... Args, typename = decltype(Transform(std::declval<Args>()...))>
```

Comment:
SFINAE to make sure the Transform is actually constructable from the pack and get a nice warning in the editor?  Can we use ```std::is_constructible```? Also, either way you need to put a comment explaining that.

Reply:
Oh that's neat I didn't know about that. I can see if that works

The reason this is needed is to let the other ctor with position and rotation work. If its not the compiler errors as it tries to always instantiate the forwarding one obv that fails since eigen has not implemented that.

Reply:
You could also use explicit specialization